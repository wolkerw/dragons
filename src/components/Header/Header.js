import React, { PureComponent } from "react";
import { Container, Row, Col } from "react-bootstrap";
import "./Header.scss";
// import logoRenner from "../../assets/images/logos/logo-lojas-renner.svg";

class Header extends PureComponent {
    render() {
        return (
            <Container fluid className="header">
                <div className="grid">
                    <Row>
                        <Col>
                            <p>Drag√µes</p>
                        </Col>
                    </Row>
                </div>
            </Container>
        );
    }
}

export default Header;
