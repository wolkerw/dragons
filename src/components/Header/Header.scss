@import "../../styles/base/base";

.header {
    height: 52px;
    background-color: $color-primary;
    position: fixed;
    color: #fff;
    z-index: 999;

    p {
        font-size: 19px;
        line-height: 22px;
        margin-left: 25px;
    }

    .grid {
        padding-top: 14px;

        .menu {
            text-align: right;
            margin-top: 6px;
        }
    }
}

.header + * {
    padding-top: 52px;
}
