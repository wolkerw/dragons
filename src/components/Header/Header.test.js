import React from "react";
import Header from "./Header";
import { IntlProvider } from "react-intl";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import messages from "../../assets/i18n";

configure({ adapter: new Adapter() });

const country = "en";

describe("Header Component", () => {
    const component = (
        <IntlProvider
            locale={country}
            key={country}
            messages={messages[{ country }]}
            defaultLocale={country}
        >
            <Header />
        </IntlProvider>
    );

    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("generated an header with a logo", () => {
        const wrapper = mount(component);
        const logo = wrapper.find("p");
        expect(logo.length).toEqual(1);
    });
});
