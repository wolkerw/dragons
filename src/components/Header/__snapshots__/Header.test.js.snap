// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches the snapshot 1`] = `
<div
  className="header container-fluid"
>
  <div
    className="grid"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <p>
          Drag√µes
        </p>
      </div>
    </div>
  </div>
</div>
`;
